### PR TITLE
Some keyword documentation.

### PR DESCRIPTION
**Pull request by gilescope**
I thought about going into detail, but I'd much rather route them to the async book asap.

### CODE REVIEW

File: src/libstd/keyword_docs.rs [hunk 1]
```diff
@@ -1149,20 +1149,27 @@ mod where_keyword {}
 //
 /// Return a [`Future`] instead of blocking the current thread.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
-///
+/// Use `async` in front of `fn`, `closure` or a `block` to turn the marked code into a `Future`.
```
Comment (Centril) [inline src/libstd/keyword_docs.rs]:
```suggestion
/// Use `async` in front of `fn`, `closure`, or a `block` to turn the marked code into a `Future`. 
```

File: src/libstd/keyword_docs.rs [hunk 2]
```diff
@@ -1149,20 +1149,27 @@ mod where_keyword {}
 //
 /// Return a [`Future`] instead of blocking the current thread.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
-///
+/// Use `async` in front of `fn`, `closure` or a `block` to turn the marked code into a `Future`. 
+/// As such the code will not be run immediately, but will only be evaluated when the returned 
+/// future is `awaited`.
```
Comment (Centril) [inline src/libstd/keyword_docs.rs]:
```suggestion
/// future is `.await`ed.
```

File: src/libstd/keyword_docs.rs [hunk 3]
```diff
@@ -1149,20 +1149,27 @@ mod where_keyword {}
...
-/// [not yet complete]: https://github.com/rust-lang/rust/issues/34601
+/// [async book]: https://rust-lang.github.io/async-book/
 mod async_keyword {}
 
 #[doc(keyword = "await")]
 //
 /// Suspend execution until the result of a [`Future`] is ready.
 ///
-/// The documentation for this keyword is [not yet complete]. Pull requests welcome!
+/// Calling `.await` on a future will halt the current function's execution until the `executor`
```
Comment (Centril) [inline src/libstd/keyword_docs.rs]:
```suggestion
/// `.await`ing a future will suspend the current function's execution until the `executor`
```

File: src/libstd/keyword_docs.rs (hunk 1 above)
Comment (gilescope) [reply src/libstd/keyword_docs.rs]: I've heard the name `oxford comma`, but never before looked up its definition. It is exactly this. Thanks - learned something new on boxing day.

**Comment by gilescope** [comment]
Have added a line on edition and stable version.

**Comment by gilescope** [comment]
( Fixes #66909 )

**Comment by Centril** [comment]
PR builder seems unhappy?

**Comment by Centril** [comment]
r=me rollup when it is happy.

**Pull request merged**